### PR TITLE
`ktlint_test` windows support

### PR DESCRIPTION
**Pull request by hanneskaeufler**
This enables windows linting support with the `ktlint_test` rule.
Previously this rule would fail such as:

```
cd examples/trivial && bazel test //... --enable_runfiles -s --test_output=all
ERROR(tools/test/windows/tw.cc:1294) ERROR: src/main/native/windows/process.cc(202): CreateProcessW("C:\users\xxxx\s7xd2g5q\execroot\__main__\bazel-out\x64_windows-fastbuild\bin\app\lint_test.runfiles\__main__\app\lint_test"): %1 ist keine zulssige Win32-Anwendung.
 (error: 193)
ERROR(tools/test/windows/tw.cc:1451) Failed to start test process (arg: C:\users\xxxx\_bazel_xxxx\s7xd2g5q\execroot\__main__\bazel-out\x64_windows-fastbuild\bin\app\lint_test.runfiles\__main__\app\lint_test)
```

Now it runs successfully. However, the actual files being linted there fail the linter due to wrong indentation.
<del>I'll probably also submit a PR that fixes indentation so we can run `test //...` on CI for the trivial examples.</del>
Included in this PR now is also the CI changes to proof that it works.

### CODE REVIEW

**Comment by Bencodes** [comment]
@hanneskaeufler can you include this in the windows CI PR so that we can validate if this works?

**Comment by hanneskaeufler** [comment]
> @hanneskaeufler can you include this in the [windows CI PR](https://github.com/bazelbuild/rules_kotlin/pull/771) so that we can validate if this works?

Yeah sure @Bencodes, I was contemplating how to do this since actually the linux example stage did not test the linter at all as well, and as mentioned in the description the file was not formatted to pass the linter. But I assume now that this was not intentional and thus enabled the tests in the example stage for both windows and linux by reformatting the file.

Enabled now and passing 🍒

File: examples/trivial/app/app.kt [hunk 1]
```diff
@@ -8,21 +8,18 @@ import graphql.GraphQL
 data class Foo(val name: String, val age: Int)
 
 class Query {
```
Comment (Bencodes) [inline examples/trivial/app/app.kt:10]: @hanneskaeufler Did you mean to pull in these changes?

Comment (hanneskaeufler) [reply examples/trivial/app/app.kt:10]: Yes, file needed to be reformatted to pass the linter

**Review by Bencodes — APPROVED** [review]

**Comment by jesseschalken** [comment]
Could the same code be used to enable Windows support for `kt_jvm_binary` and `kt_jvm_test`?

**Comment by hanneskaeufler** [comment]
Probably something along those lines should work, I will be looking into that over the coming week(s). I just started with the linter because that was easiest to get into the codebase 😂

**Pull request merged**